packages/manager/apps/pci-ai-tools/src/components/order/app-scaling/AutoScalingForm/AutoScalingForm.component.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useForm, useWatch } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Form } from '@datatr-ux/uxlib';
+
+import Price from '@/components/price/Price.component';
+import { AppPricing, Scaling } from '@/types/orderFunnel';
+import {
+  createScalingSchema,
+  getInitialValues,
+  toScaling,
+  ScalingFormValues,
+} from '@/lib/scalingHelper';
+import { ReplicaFields } from './ReplicaFields';
+import { ResourceTypeSelector } from './ResourceTypeSelector';
+import { CpuRamFields } from './CpuRamFields';
+import { CustomMetricsFields } from './CustomMetricsFields';
+
+interface AutoScalingFormProps {
+  scaling: Scaling;
+  onChange: (scalingStrat: Scaling) => void;
+  pricingFlavor?: AppPricing;
+  onNonValidForm?: (isPending: boolean) => void;
+}
+
+export const AutoScalingForm = React.forwardRef<
+  HTMLDivElement,
+  AutoScalingFormProps
+>(({ scaling, onChange, pricingFlavor, onNonValidForm }, ref) => {
+  const { t } = useTranslation('ai-tools/components/scaling');
+
+  const schema = useMemo(() => createScalingSchema(t), [t]);
+
+  const form = useForm<ScalingFormValues>({
+    resolver: zodResolver(schema),
+    defaultValues: getInitialValues(scaling),
+    mode: 'onChange',
+  });
+
+  const { control, setValue, getValues } = form;
+
+  const resType = useWatch({ control, name: 'resType' });
+  const minRep = useWatch({ control, name: 'minRep' });
+
+  const isCustom = resType === 'CUSTOM';
+
+  const notify = useCallback(() => {
+    const values = getValues();
+    const isValid = schema.safeParse(values).success;
+
+    onChange(toScaling(scaling, values));
+    onNonValidForm?.(!isValid);
+  }, [getValues, onChange, onNonValidForm, scaling, schema]);
```

Comment:
You are duplicating react-hook-form here, it handles validation and error messages on itself